### DISCUSSION TITLE
Eslint support? (NodeJS)

### DISCUSSION THREAD

**Discussion by mandrasch** [php-perfect/ddev-intellij-plugin#235] [Q&A]
Hey, 

just wanted to ask if eslint fully works via this (awesome) integration plugin in PhpStorm?

As far as I understand it runs via the NodeJS interpreter 
https://www.jetbrains.com/help/phpstorm/eslint.html#ws_node_docker_lint

I see this:

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/2707016b-d5d4-41b2-9fd9-70cdad6fcae0)

But also get this, since I'm using colima:

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/9dccd6f1-3bb0-4d4c-bd3b-49288fdcf1d9)

https://github.com/php-perfect/ddev-intellij-plugin/issues/83 (Don't know if this is related)

Config:

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/7cde47cd-f984-4357-b6ff-330adf97ae14)

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/74592223-9741-49e0-93f8-e648474d3399)

I'm running on
- colima version 0.5.5
- ddev version v1.22.1
- OSX 13.5.2
- PhpStorm: 2023.2.1

Thanks very much for information and thanks very much for providing this plugin!

Cheers!

**Comment by mandrasch** [comment]
Okay turns out this works, but is not a good idea (yet)

When a TS file is open(!), Eslint service will try to connect & starts web containers, interfering with `ddev start` which can lead to timeout issues. and it really slows down your phpstorm, at least thats what I was experiencing.

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/4debe152-7e44-4503-8e4f-ce2f3404b973)

In the meantime I switched  to local node again

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/e4aa2ab7-cd1c-49dd-ad99-07fcf460d587)


More info: https://discord.com/channels/664580571770388500/1161952380062011422

Hi @nico-loeber, I just had one question regarding this - what is does this automatic configuration "nodejs interpreter" mean? And what settings should I see in NodeJS interpreter settings? Or is there an override? 

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/777278/b8107411-9d52-49ca-9e58-23e827e3edde)

thanks for providing the ddev plugin!

**Comment by nico-loeber** [comment]
Hey @mandrasch,
the plugin does provide a node interpreter configuration. You should find it in `File | Settings | Languages & Frameworks | Node.js`.
This interpreter configuration is used for running yarn / npm and for executing other node based tooling you might use in your frontend workflow. There are currently some limitations that might include eslint. See https://github.com/php-perfect/ddev-intellij-plugin/issues/156 for more detail.
The interpreter is configured to use the existing and running ddev container, so it should not start a new container like you described.
The current limitations come from PHPStorm not fully supporting remote node interpreters, so for now you might need to use a local installation and work with that one until the support is improved.

You can check if you make it to configure PHPStorm to use the ddev container for eslint manually. If this works, we can then do this configuration automatically by the plugin.
Otherwise I would like to encourage you to open a ticket for Webstorm/PHPStorm.

**Reply by rfay** [reply]
The thing is here @nico-loeber that it's creating a large bug that the ddev-intellij-plugin is adding a nodejs configuration that uses the container-based node. I recommend that until Jetbrains supports "exec" for node (and several other things) that this plugin not create that setup, as it can do nothing but trouble. There are a few existing issues about this in youtrack, for example https://youtrack.jetbrains.com/issue/WEB-53172

**Comment by iamntz** [comment]
hey guys, i figured out something that may help others too: make Node and phpstorm work together (though I'm not sure how it plays with eslint, but it worth a try)

some while ago i was very confused about how should phpstorm be configured in order to have node inside docker. unfortunately, phpstorm is kind of limited in this regards (i.e. doesn't allow to reuse a container, but only to start it) and (probably) it will take some years to have it fixed. so here is a little workaround: add a new node remote interpreter with node:18-alpine as image name (or whatever version you need).

this won't use the ddev version but it will use a docker container. with a bit of a trickery, you could use ddev npm for managing packages and this remote interpreter for... stuff.

it took me a while to figure out that this is possible, i'm sure it may have some limitations, but for what i need it is good enough.

![image](https://github.com/php-perfect/ddev-intellij-plugin/assets/132062/231fc81f-88b9-44dc-a27c-5d972277dedc)

discussion also started [here](https://discord.com/channels/664580571770388500/1068592266752573440/1166615613658120202)

**Reply by mandrasch** [reply]
Hey! thanks for sharing!  nice workaround if I don't want to use the nodejs version on my host computer / switch faster between node versions 👍 if I have time in the next weeks I'll check with eslint.

**Reply by iamntz** [reply]
@mandrasch my goal was to have the the ability to debug some scripts, so I have no idea how this works for eslint. 

But for debugging, it works :)

**Reply by iamntz** [reply]
Update: I tested and I can confirm that the method above works for eslint _and_ prettier.
